client/landing/jetpack-cloud/components/scan-history-item/index.jsx

```diff
@@ -25,61 +28,216 @@ class ScanHistoryItem extends Component {
+							} ) }
+						</span>
+					) }
 				</div>
 				<Badge
 					className={ classnames(
 						'scan-history-item__badge',
 						this.entryActionClassNames( entry )
 					) }
 				>
-					<small>{ entry.action }</small>
+					<small>{ entry.status }</small>
 				</Badge>
 			</>
 		);
 	}
 
+	getDetailType( threat ) {
+		if ( threat.hasOwnProperty( 'diff' ) ) {
+			return 'core';
+		}
+
+		if ( threat.hasOwnProperty( 'context' ) ) {
+			return 'file';
+		}
+
+		if ( threat.hasOwnProperty( 'extension' ) ) {
+			// 'plugin' or 'theme'
+			return threat.extension.type;
+		}
+
+		if ( threat.hasOwnProperty( 'rows' ) ) {
+			return 'database';
+		}
+
+		if ( 'Suspicious.Links' === threat.signature ) {
+			return 'database';
+		}
+
+		return 'none';
+	}
+
+	getSubtitle( threat ) {
+		switch ( this.getDetailType( threat ) ) {
+			case 'core':
+				return translate( 'Vulnerability found in WordPress' );
+
+			case 'file':
+				return translate( 'Threat found ({{signature/}})', {
+					components: {
+						signature: <span className="threat-item__alert-signature">{ threat.signature }</span>,
+					},
+				} );
+
+			case 'plugin':
+				return translate( 'Vulnerability found in plugin' );
+
+			case 'theme':
+				return translate( 'Vulnerability found in theme' );
+
+			case 'database':
+				return null;
+
+			case 'none':
+			default:
+				return translate( 'Miscellaneous vulnerability' );
+		}
+	}
+
+	getTitle( threat ) {
+		// This should be temprary since this data should be coming from the api
+		// and not something that we should change to accompadate the results.
+		const { filename, extension } = threat;
+
+		const basename = s => s.replace( /.*\//, '' );
+
+		switch ( this.getDetailType( threat ) ) {
+			case 'core':
+				return translate( 'Infected core file: {{filename/}} ', {
+					components: {
+						filename: <code className="threat-item__alert-filename">{ basename( filename ) }</code>,
+					},
+				} );
+
+			case 'file':
+				return translate( 'The file {{filename/}} contains a malicious code pattern.', {
+					components: {
+						filename: <code className="threat-item__alert-filename">{ basename( filename ) }</code>,
+					},
+				} );
+
+			case 'plugin':
+				return translate( 'Vulnerable Plugin: {{pluginSlug/}} (version {{version/}})', {
+					components: {
+						pluginSlug: <span className="threat-item__alert-slug">{ extension.slug }</span>,
+						version: <span className="threat-item__alert-version">{ extension.version }</span>,
+					},
+				} );
+
+			case 'theme':
+				return translate( 'Vulnerable Theme {{themeSlug/}} (version {{version/}})', {
+					components: {
+						themeSlug: <span className="threat-item__alert-slug">{ extension.slug }</span>,
+						version: <span className="threat-item__alert-version">{ extension.version }</span>,
+					},
+				} );
+
+			case 'database':
+				return translate( 'Database %(threatCount)d threat', 'Database %(threatCount)d threats', {
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Jetpack identified %(threatCount)d threat in your database.', 'Jetpack identified %(threatCount)d threats in your database.' )` *ES Score: 7*